src/App.tsx

```diff
@@ -1,73 +1,109 @@
-import React from 'react';
+import { useState, useRef } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import debounce from 'lodash.debounce';
+import { Autocomplete } from './components/autocomplete';
+import { NoMatch } from './components/nomatch';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const initialPerson: Person = {
+    name: '',
+    born: 0,
+    died: 0,
+    slug: '',
+    fatherName: '',
+    motherName: '',
+    sex: 'm',
+  };
+
+  const [selectedPerson, setSelectedPerson] = useState<Person>(initialPerson);
+  const [isDropdownActive, setIsDropdownActive] = useState(false);
+  const [suggestions, setSuggestions] = useState<Person[]>(peopleFromServer);
+
+  const { name, born, died } = selectedPerson;
+
+  const titleText =
+    selectedPerson.name !== initialPerson.name
+      ? `${name} (${born} - ${died})`
+      : 'No selected person';
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const handleDropdownClick = (a: boolean) => {
+    setTimeout(() => {
+      setIsDropdownActive(a);
+    }, 200);
```

Comment:
I guess the timeout is redundant in this case, so you can use `setIsDropdownActive` instead of `handleDropdownClick` below

Reply:
Need this timeout to rewrite selected person